src/python/pants/backend/python/goals/debug_goals.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import json
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Optional, Union
+
+from pants.backend.project_info.peek import _PeekJsonEncoder
+from pants.backend.python.dependency_inference.module_mapper import ResolveName
+from pants.backend.python.dependency_inference.parse_python_dependencies import (
+    ParsedPythonDependencies,
+    ParsedPythonImportInfo,
+)
+from pants.backend.python.dependency_inference.rules import (
+    ImportResolveResult,
+    PythonImportDependenciesInferenceFieldSet,
+    PythonInferSubsystem,
+    ResolvedParsedPythonDependencies,
+    ResolvedParsedPythonDependenciesRequest,
+    UnownedImportsPossibleOwners,
+    UnownedImportsPossibleOwnersRequest,
+    _collect_imports_info,
+    _exec_parse_deps,
+    _find_other_owners_for_unowned_imports,
+)
+from pants.backend.python.goals.run_python_source import PythonSourceFieldSet
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.build_graph.address import Address
+from pants.engine.console import Console
+from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.internals.selectors import Get, MultiGet
+from pants.engine.rules import collect_rules, goal_rule, rule
+from pants.engine.target import Targets
+from pants.option.option_types import EnumOption
+
+
+class AnalysisFlavor(Enum):
```

Comment:
I'm open to better names

Reply:
I'm not too concerned with the name, but a bit longer help string on the option would be good.